compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -106,6 +109,9 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
         if (tp.cls.isAnonymousClass) keywordStr("this")
         if (tp.cls.is(ModuleClass)) fullNameString(tp.cls.sourceModule)
         else super.toTextRef(tp)
+      case tp: TermRef if !printDebug =>
+        if tp.symbol.is(Package) then fullNameString(tp.symbol)
```

Comment:
This caused a bit of regression in metals since we are no longer able to omit a package when printing : https://github.com/scalameta/metals/pull/5327

I wonder if you could do a bit more full proof mechanism here so that we always correctly skip prefixes in both metals and compiler :thinking:  It's influenced by the imports within metals.